### ISSUE TITLE
rocm-smi parsing: GPU name and VRAM misdetected for discrete RDNA4 (RX 9060 XT)

### ISSUE THREAD

**Issue by cyrilvoynov** [Pavelevich/llm-checker#79]
## Summary

Two regex bugs in `src/hardware/backends/rocm-detector.js` (method `_detectViaRocmSmi`) cause the detector to misreport name and VRAM for a discrete RDNA 4 GPU (AMD Radeon RX 9060 XT, device `0x7590`, `gfx1200`, 16 GB).

As a result the detected GPU object becomes:

```json
{
  "name": "GFX Version: \t\tgfx1200",
  "memory": { "total": 2, "free": 2, "used": 0 },
  "speedCoefficient": 40
}
```

and `llm-checker check` downgrades recommendations to 1B models despite 16 GB of VRAM being physically available.

## Environment

- `llm-checker` 3.5.11 (installed via npm)
- CPU: AMD Ryzen 7 7800X3D
- GPU: AMD Radeon RX 9060 XT 16 GB (Navi 44, RDNA 4, device `0x7590`, `gfx1200`)
- OS: Linux (CachyOS, kernel 7.0.0)
- ROCm tools: `rocm-smi` reports version `4.0.0`

## Actual `rocm-smi` output

```
$ rocm-smi --showproductname
GPU[0]		: Card Series: 		AMD Radeon RX 9060 XT
GPU[0]		: Card Model: 		0x7590
GPU[0]		: Card Vendor: 		Advanced Micro Devices, Inc. [AMD/ATI]
GPU[0]		: Card SKU: 		P816G493
GPU[0]		: Subsystem ID: 	-0x1b6d
GPU[0]		: Device Rev: 		0xc0
GPU[0]		: Node ID: 		1
GPU[0]		: GUID: 		2347
GPU[0]		: GFX Version: 		gfx1200

$ rocm-smi --showmeminfo vram
GPU[0]		: VRAM Total Memory (B): 17095983104
GPU[0]		: VRAM Total Used Memory (B): 1729613824
```

## Bug 1 — GPU name parsing

`src/hardware/backends/rocm-detector.js:240`:

```js
const nameMatches = gpuList.matchAll(/GPU\[(\d+)\].*?:\s*(.+)/g);
for (const match of nameMatches) {
    gpuNames[parseInt(match[1])] = match[2].trim();
}
```

The regex has no anchor for \"Card Series\", so every `GPU[0]: <anything>: <value>` line matches and overwrites the previous one. The last match (`GFX Version: gfx1200`) wins, so `name` ends up as `\"GFX Version: \\t\\tgfx1200\"`.

**Fix:** anchor to `Card Series` specifically.

## Bug 2 — VRAM parsing

`src/hardware/backends/rocm-detector.js:256`:

```js
const lineMatch = line.match(/GPU\[(\d+)\].*?Total.*?Memory\s*(?:\(([^)]+)\))?\s*:\s*(\d+)/i);
```

`.*?Total.*?Memory` matches **both** lines:
- `VRAM Total Memory (B): 17095983104` → 16 GB
- `VRAM Total Used Memory (B): 1729613824` → 2 GB

The second match overwrites `gpuMemory[0]`, leaving `total: 2`. For a 16 GB card currently using ~1.6 GB, the detector reports 2 GB total.

**Fix:** skip lines containing \"Total Used Memory\", or anchor the regex to `VRAM Total Memory` without the `Used` variant.

## Bug 3 (related) — integer-like keys in `speedMap`

In `calculateSpeedCoefficient` the `speedMap` is an object. JS engines iterate integer-like string keys in numeric order **before** other keys, regardless of insertion order. This means if someone adds keys like `'9060'` alongside `'9060 xt'`, the iteration order becomes:

```
[ '9060', '9070', 'r9700', 'ai pro r9700', '9070 xt', '9060 xt' ]
```

and the more specific `'9060 xt'` branch is never reached for `\"AMD Radeon RX 9060 XT\"` (the bare `'9060'` matches first).

The existing keys in the codebase all contain non-digit characters (e.g. `'6600 xt'`, `'7900 xtx'`), so they are safe today — but this is a sharp edge for anyone extending the map. Suggest either using a `Map`, or prefixing numeric keys (e.g. `'rx 9060 xt'`).

## Proposed patch

```diff
--- a/src/hardware/backends/rocm-detector.js
+++ b/src/hardware/backends/rocm-detector.js
@@ -19,6 +19,9 @@ class ROCmDetector {
     static AMD_DEVICE_IDS = {
         // RDNA 4 / Radeon AI PRO
         '7551': { name: 'AMD Radeon AI PRO R9700', vram: 32 },
+        '7590': { name: 'AMD Radeon RX 9060 XT', vram: 16 },
+        '7580': { name: 'AMD Radeon RX 9070 XT', vram: 16 },
+        '7581': { name: 'AMD Radeon RX 9070', vram: 16 },
         // RDNA 3 (RX 7000 series)
         '744c': { name: 'AMD Radeon RX 7900 XTX', vram: 24 },
@@ -237,9 +240,10 @@
-            // Parse GPU names
+            // Parse GPU names — match only \"Card Series\" to avoid being
+            // overwritten by later fields (Card Model, GFX Version, etc.).
             const gpuNames = [];
-            const nameMatches = gpuList.matchAll(/GPU\\[(\\d+)\\].*?:\\s*(.+)/g);
+            const nameMatches = gpuList.matchAll(/GPU\\[(\\d+)\\]\\s*:\\s*Card Series\\s*:\\s*(.+)/g);
             for (const match of nameMatches) {
                 gpuNames[parseInt(match[1])] = match[2].trim();
             }
@@ -253,7 +257,9 @@
             const gpuMemory = {};
             const memLines = String(memInfo || '').split('\\n');
             for (const line of memLines) {
-                const lineMatch = line.match(/GPU\\[(\\d+)\\].*?Total.*?Memory\\s*(?:\\(([^)]+)\\))?\\s*:\\s*(\\d+)/i);
+                // Skip \"Total Used Memory\" — we only want the total capacity.
+                if (/Total Used Memory/i.test(line)) continue;
+                const lineMatch = line.match(/GPU\\[(\\d+)\\].*?Total Memory\\s*(?:\\(([^)]+)\\))?\\s*:\\s*(\\d+)/i);
                 if (!lineMatch) continue;
```

Plus name/VRAM/speed entries for the RX 9000 series in `getGPUCapabilities`, `estimateVRAMFromModel`, and `calculateSpeedCoefficient` (with `'rx '` prefixes in `speedMap` to avoid the integer-key ordering pitfall).

Verified locally: after the patch the detector correctly reports

```json
{
  "name": "AMD Radeon RX 9060 XT",
  "memory": { "total": 16 },
  "capabilities": { "architecture": "RDNA 4", "gfxVersion": "gfx1200" },
  "speedCoefficient": 170,
  "totalVRAM": 16
}
```

Happy to open a PR if useful.
